libsplinter/src/auth/rest_api/identity/openid.rs

```diff
@@ -56,12 +56,13 @@ impl IdentityProvider for OpenIdUserIdentityProvider {
             }
         }
 
-        let email = response
+        let user_info = response
             .json::<UserResponse>()
-            .map_err(|_| InternalError::with_message("Received unexpected response body".into()))?
-            .email;
+            .map_err(|_| InternalError::with_message("Received unexpected response body".into()))?;
 
-        Ok(Some(email))
+        let user_identity = user_info.clone().email.or(user_info.name);
```

Comment:
This clone doesn't seem necessary.  You should be able to destructure `user_info` and just use the `email` and `name` directly.

Reply:
Fixed, the `clone` was needed when `.or_else()` was used here but isn't with `.or()`